dags/elsevier/repository.py

```diff
@@ -0,0 +1,61 @@
+import os
+from io import BytesIO
+from typing import IO
+
+from common.repository import IRepository
+from common.s3_service import S3Service
+from common.utils import find_extension
+
+
+class ElsevierRepository(IRepository):
+    ZIPED_DIR: str = "raw/"
+    EXTRACTED_DIR: str = "extracted/"
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.s3 = S3Service(os.getenv("ELSEVIER_BUCKET_NAME", "elsevier"))
+
+    def get_all_raw_filenames(self):
+        return [
+            f.key.replace("raw/", "")
+            for f in self.s3.objects.filter(Prefix=self.ZIPED_DIR).all()
+        ]
+
+    def find_all(self, filenames_to_process=None):
+        grouped_files = {}
```

Comment:
I think you need a better name for this dict (might be longer but more meaningful), it's confusing

Reply:
for this one, I might make a new PR. If I will change it for Eslevier, I would prefer also change in all other publishers